invenio/modules/search/views/search.py

```diff
@@ -462,7 +462,7 @@ def search(collection, p, of, ot, so, rm):
 
     ctx = dict(
         facets=facets.get_facets_config(collection, qid),
-        records=len(get_current_user_records_that_can_be_displayed(qid)),
+        records=len(recids),
```

Comment:
@jirikuncar FYI we were debugging this with @remileduc the other day. The behaviour looks OK.  The `get_current_user_records_that_can_be_displayed(qid)` bit seems to have been there since a long time.  Do you remember why `recids` is not reused for the search hit count context even though it is used "as is" for record printing later?